SharedContent/cpp-winui-template/WinUI3TemplateCpp/SettingsPage.xaml.cpp

```diff
@@ -0,0 +1,77 @@
+﻿#include "pch.h"
+#include "SettingsPage.xaml.h"
+#if __has_include("SettingsPage.g.cpp")
+#include "SettingsPage.g.cpp"
+#endif
+#include <SampleConfiguration.h>
+
+using namespace winrt;
+using namespace Microsoft::UI::Xaml;
+using namespace Microsoft::UI::Xaml::Controls;
+
+namespace winrt::WinUI3TemplateCpp::implementation
+{
+    SettingsPage::SettingsPage()
+    {
+        InitializeComponent();
+    }
+
+    void SettingsPage::OnNavigatedTo(Microsoft::UI::Xaml::Navigation::NavigationEventArgs const& e)
+    {
+        // Initialize the CurrentTheme on first navigation to the Settings page
+        if (winrt::WinUI3TemplateCpp::Settings::CurrentTheme == L"")
+        {
+            if (RequestedTheme() == ElementTheme::Dark) 
+            {
+                Settings::CurrentTheme = L"Dark";
```

Comment:
Not a fan of using a string to record the theme. Can we just make the tag an ElementTheme?
```xml
<RadioButton Checked="OnThemeRadioButtonChecked" Content="Dark">
    <RadioButton.Tag><ElementTheme>Dark</ElementTheme></RadioButton.Tag>
</RadioButton>
```

Then we would do `c.as<RadioButton>().Tag().as<ElementTheme>()`.

If we like strings, we should at least convert the string to an ElementTheme immediately upon parsing the tag, and then work in ElementTheme from then on.

```cpp
ElementTheme ParseElementTheme(hstring const& s)
{
    if (s == L"Light") return ElementTheme::Light;
    if (s == L"Dark") return ElementTheme::Dark;
    return ElementTheme::Default;
}
```

Reply:
Changed to use `ElementTheme` instead of strings.